examples/app/headless_renderer.rs

```diff
@@ -0,0 +1,547 @@
+                info!("Adding CaptureFramePlugin");
+                app.add_systems(PostUpdate, update);
+            }
+        }
+
+        #[derive(Debug, Default, Resource)]
+        pub struct SceneController {
+            state: SceneState,
+            name: String,
+            width: u32,
+            height: u32,
+            single_image: bool,
+        }
+
+        impl SceneController {
+            pub fn new(width: u32, height: u32, single_image: bool) -> SceneController {
+                SceneController {
+                    state: SceneState::BuildScene,
+                    name: String::from(""),
+                    width,
+                    height,
+                    single_image,
+                }
+            }
+        }
+
+        #[derive(Debug, Default)]
+        pub enum SceneState {
+            #[default]
+            BuildScene,
+            Render(u32),
+        }
+
+        pub fn setup_render_target(
+            commands: &mut Commands,
+            images: &mut ResMut<Assets<Image>>,
+            render_device: &Res<RenderDevice>,
+            scene_controller: &mut ResMut<SceneController>,
+            pre_roll_frames: u32,
+            scene_name: String,
+        ) -> RenderTarget {
+            let size = Extent3d {
+                width: scene_controller.width,
+                height: scene_controller.height,
+                ..Default::default()
+            };
+
+            // This is the texture that will be rendered to.
+            let mut render_target_image = Image {
+                texture_descriptor: TextureDescriptor {
+                    label: None,
+                    size,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::Rgba8UnormSrgb,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    usage: TextureUsages::COPY_SRC
+                        | TextureUsages::COPY_DST
+                        | TextureUsages::TEXTURE_BINDING
+                        | TextureUsages::RENDER_ATTACHMENT,
+                    view_formats: &[],
+                },
+                ..Default::default()
+            };
+            render_target_image.resize(size);
+            let render_target_image_handle = images.add(render_target_image);
+
+            // This is the texture that will be copied to.
+            let mut cpu_image = Image {
+                texture_descriptor: TextureDescriptor {
+                    label: None,
+                    size,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::Rgba8UnormSrgb,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    usage: TextureUsages::COPY_DST | TextureUsages::TEXTURE_BINDING,
+                    view_formats: &[],
+                },
+                ..Default::default()
+            };
+            cpu_image.resize(size);
```

Comment:
This can be simplified a little with this:

```suggestion
        let mut cpu_image = Image::new_fill(
            image_size,
            TextureDimension::D2,
            &[0; 4],
            TextureFormat::bevy_default(),
            RenderAssetUsages::default(),
        );
```